### ISSUE TITLE
Make import sorting/aligning optional

### ISSUE THREAD

**Issue by ivan-m** [haskell/haskell-mode#914]
I use `stylish-haskell` to sort and align my imports; it's thus visually confusing when I add an import, haskell-mode sorts and aligns it (undoing some of the layout that stylish-haskell does), and then as soon as I save the layout changes again.

**Comment by gracjan** [comment]
What function does the sorting?

**Comment by ivan-m** [comment]
`haskell-add-import` calls `haskell-sort-imports` and `haskell-align-imports`.

**Comment by gracjan** [comment]
Ah, I see. It should:
1. Find out if current import list is sorted.
2. If so keep the sorting order.
3. Otherwise just add import at the end of the list.

That way we can get away without a setting with a 'just works!' functionality. Somebody needs to implement the 'is sorted' function.

**Comment by ivan-m** [comment]
Or, have a customisation option as to whether it *should* do any sorting/aligning.

**Comment by gracjan** [comment]
Customization options present a usability challenge. They are a testing nightmare and require users to discover the option even exists.

**Comment by ivan-m** [comment]
Did you see my update? (I did an edit of my previous comment right after I posted it.)

This of course assumes that the stylish-haskell config doesn't have `imports -> align -> none` (I think that covers sorting as well).

The other possible solution - and why I see the noticeable changes - is that the sorting and aligning in haskell-mode doesn't seem to have anything similar to stylish-haskell's `group` option; I'm not sure if this could be done automagically in haskell-mode unless you tried to detect whether the various import blocks are already indented or not.
